util/vec-map/src/lib.rs

```diff
@@ -0,0 +1,300 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+//! Provides a map (key-value store) interface backed by `heapless::Vec`
+
+#![deny(missing_docs)]
+
+use displaydoc::Display;
+use heapless::Vec;
+
+/// An error which can occur when using VecMap
+#[derive(Clone, Debug, Display, Eq, PartialEq)]
+pub enum Error {
+    /// VecMap capacity exceeded
+    CapacityExceeded,
+}
+
+/// This is a mini version of VecMap that is no_std compatible and uses
+/// heapless::Vec instead alloc::Vec.
+///
+/// It may be better to patch upstream for no_std compatibility and use that,
+/// but that crate has other issues -- it relies on an
+/// experimental "contracts" crate that causes a dependency on rand crate.
+/// Porting to Heapless would be a breaking change to the API.
+///
+/// TBD what the best path is: https://github.com/p-avital/vec-map-rs/blob/master/src/lib.rs
+#[derive(Clone, Debug)]
+pub struct VecMap<K, V, const N: usize> {
+    keys: Vec<K, N>,
+    values: Vec<V, N>,
+}
+
+impl<K, V, const N: usize> Default for VecMap<K, V, N> {
+    fn default() -> Self {
+        Self {
+            keys: Default::default(),
+            values: Default::default(),
+        }
+    }
+}
+
+impl<K, V, const N: usize> VecMap<K, V, N> {
+    /// Check if the map is empty
+    #[inline]
+    pub fn is_empty(&self) -> bool {
+        self.len() == 0
+    }
+
+    /// Get the number of items in the map
+    #[inline]
+    pub fn len(&self) -> usize {
+        debug_assert!(self.keys.len() == self.values.len());
+        self.keys.len()
+    }
+}
+
+impl<K: Eq + PartialEq, V, const N: usize> VecMap<K, V, N> {
```

Comment:
Nit: `Eq` extends `PartialEq`
```suggestion
impl<K: Eq, V, const N: usize> VecMap<K, V, N> {
```